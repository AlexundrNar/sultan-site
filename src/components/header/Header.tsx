import React, { FC } from "react";
import { useNavigate } from "react-router-dom";
import { useAppSelector } from "../../store/hooks/redux";

import "./header.scss";

const Header: FC = () => {
  const navigate = useNavigate();
  const totalQuantity = useAppSelector((state) => state.cartSlice.totalQuantity);
  const totalAmount = useAppSelector((state) => state.cartSlice.totalAmount);

  return (
    <header>
      <div className="header__top">
        <div className="container">
          <div className="nav">
            <div className="nav__address">
              <img src={`${process.env.PUBLIC_URL}/images/marker.svg`} alt="" />
              <div>
                <h3>г. Кокчетав, ул. Ж. Ташенова 129Б</h3>
                <span>(Рынок Восточный)</span>
              </div>
            </div>
            <div className="nav__contact">
              <img src={`${process.env.PUBLIC_URL}/images/convert.svg`} alt="" />
              <div>
                <h3>opt.sultan@mail.ru</h3>
                <span>На связи в любое время</span>
              </div>
            </div>
            <div className="nav__info">
              <div>
                <p>О компании</p>
              </div>
              <div>
                <p>Доставка и оплата</p>
              </div>
              <div>
                <p>Возврат</p>
              </div>
              <div>
                <p>Контакты</p>
              </div>
            </div>
          </div>
        </div>
      </div>

      <div className="header__bot">
        <div className="container">
          <div className="header__content">
            <span className="burger">
              <img src={`${process.env.PUBLIC_URL}/images/burger.svg`} alt="" />
            </span>
            <img className="rgb" src={`${process.env.PUBLIC_URL}/images/RGB.svg`} alt="" />
            <button className="header__catalog" onClick={() => navigate("/")}>
              <img className="sqg" src={`${process.env.PUBLIC_URL}/images/squares_grey.svg`} alt="" />
              Каталог
              <img className="sq" src={`${process.env.PUBLIC_URL}/images/squares.svg`} alt="" />
            </button>
            <div className="header__search">
              <input type="text" placeholder="Поиск..." />
              <button>
                <img className="srg" src={`${process.env.PUBLIC_URL}/images/search_grey.svg`} alt="" />
                <img className="sr" src={`${process.env.PUBLIC_URL}/images/search.svg`} alt="" />
              </button>
            </div>
            <div className="header__tel">
              <h3>+7 (777) 490-00-91</h3>
              <p>время работы: 9:00-20:00</p>
              <span>Заказать звонок</span>
            </div>
            <img className="header__woman" src={`${process.env.PUBLIC_URL}/images/call_woman.png`} alt="" />
            <button className="header__price" onClick={() => navigate("/admin")}>
              АДМИН-ПАНЕЛЬ
              <img src={`${process.env.PUBLIC_URL}/images/arrow_down.svg`} alt="" />
            </button>
            <div className="header__cart">
              <img
                src={`${process.env.PUBLIC_URL}/images/cart.svg`}
                alt=""
                onClick={() => navigate("/checkout")}
              />
              <span data-testid="количество">{totalQuantity}</span>
            </div>
            <div className="header__cart-check">
              <p>Корзина</p>
              <span data-testid="сумма">{totalAmount} ₸</span>
            </div>
          </div>
        </div>
      </div>
    </header>
  );
};

export default Header;
