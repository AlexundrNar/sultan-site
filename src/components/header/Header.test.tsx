const mockedUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
   ...jest.requireActual('react-router-dom') as any,
  useNavigate: () => mockedUsedNavigate,
}));

jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux") as any,
  useSelector: jest.fn(),
  useDispatch: jest.fn()
}));

import Header from './Header';
import * as ReactDom from "react-dom"
import * as reduxHooks from "react-redux"
import { fireEvent, render, screen } from '@testing-library/react';


describe('header', () => {
  let container: HTMLElement

  beforeAll(() => {
    container = document.createElement('header')
    document.body.appendChild(container)
    ReactDom.render(<Header />, container)
  })

  afterAll(() => {
    document.body.removeChild(container)
    container.remove()
  })

  it('render header', () => {
    const component = render(<Header />)
    expect(component).toMatchSnapshot()
  })

  it('inputs in header', () => {
    const inputs = container.querySelectorAll('input')
    expect(inputs).toHaveLength(1)
  })

  it('admin btn check', () => {
    const adminBtn = screen.getByText(/АДМИН-ПАНЕЛЬ/i)
    expect(adminBtn).toBeInTheDocument()
  })

  it('admin btn click', () => {
    const adminBtn = screen.getByText(/АДМИН-ПАНЕЛЬ/i)
    fireEvent.click(adminBtn)
    expect(mockedUsedNavigate).toHaveBeenCalled()
  })

  it('admin btn check', () => {
    const qnty = screen.getByTestId('количество')
    const amount = screen.getByTestId('сумма')

    jest.spyOn(reduxHooks, 'useSelector').mockReturnValue(qnty)
    jest.spyOn(reduxHooks, 'useSelector').mockReturnValue(amount)
    
    expect(qnty).toBeInTheDocument()
    expect(amount).toBeInTheDocument()
  })

})