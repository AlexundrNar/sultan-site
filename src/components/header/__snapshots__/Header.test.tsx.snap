// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header render header 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <header>
      <header>
        <div
          class="header__top"
        >
          <div
            class="container"
          >
            <div
              class="nav"
            >
              <div
                class="nav__address"
              >
                <img
                  alt=""
                  src="/images/marker.svg"
                />
                <div>
                  <h3>
                    г. Кокчетав, ул. Ж. Ташенова 129Б
                  </h3>
                  <span>
                    (Рынок Восточный)
                  </span>
                </div>
              </div>
              <div
                class="nav__contact"
              >
                <img
                  alt=""
                  src="/images/convert.svg"
                />
                <div>
                  <h3>
                    opt.sultan@mail.ru
                  </h3>
                  <span>
                    На связи в любое время
                  </span>
                </div>
              </div>
              <div
                class="nav__info"
              >
                <div>
                  <p>
                    О компании
                  </p>
                </div>
                <div>
                  <p>
                    Доставка и оплата
                  </p>
                </div>
                <div>
                  <p>
                    Возврат
                  </p>
                </div>
                <div>
                  <p>
                    Контакты
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="header__bot"
        >
          <div
            class="container"
          >
            <div
              class="header__content"
            >
              <span
                class="burger"
              >
                <img
                  alt=""
                  src="/images/burger.svg"
                />
              </span>
              <img
                alt=""
                class="rgb"
                src="/images/RGB.svg"
              />
              <button
                class="header__catalog"
              >
                <img
                  alt=""
                  class="sqg"
                  src="/images/squares_grey.svg"
                />
                Каталог
                <img
                  alt=""
                  class="sq"
                  src="/images/squares.svg"
                />
              </button>
              <div
                class="header__search"
              >
                <input
                  placeholder="Поиск..."
                  type="text"
                />
                <button>
                  <img
                    alt=""
                    class="srg"
                    src="/images/search_grey.svg"
                  />
                  <img
                    alt=""
                    class="sr"
                    src="/images/search.svg"
                  />
                </button>
              </div>
              <div
                class="header__tel"
              >
                <h3>
                  +7 (777) 490-00-91
                </h3>
                <p>
                  время работы: 9:00-20:00
                </p>
                <span>
                  Заказать звонок
                </span>
              </div>
              <img
                alt=""
                class="header__woman"
                src="/images/call_woman.png"
              />
              <button
                class="header__price"
              >
                АДМИН-ПАНЕЛЬ
                <img
                  alt=""
                  src="/images/arrow_down.svg"
                />
              </button>
              <div
                class="header__cart"
              >
                <img
                  alt=""
                  src="/images/cart.svg"
                />
                <span
                  data-testid="количество"
                />
              </div>
              <div
                class="header__cart-check"
              >
                <p>
                  Корзина
                </p>
                <span
                  data-testid="сумма"
                >
                   ₸
                </span>
              </div>
            </div>
          </div>
        </div>
      </header>
    </header>
    <div>
      <header>
        <div
          class="header__top"
        >
          <div
            class="container"
          >
            <div
              class="nav"
            >
              <div
                class="nav__address"
              >
                <img
                  alt=""
                  src="/images/marker.svg"
                />
                <div>
                  <h3>
                    г. Кокчетав, ул. Ж. Ташенова 129Б
                  </h3>
                  <span>
                    (Рынок Восточный)
                  </span>
                </div>
              </div>
              <div
                class="nav__contact"
              >
                <img
                  alt=""
                  src="/images/convert.svg"
                />
                <div>
                  <h3>
                    opt.sultan@mail.ru
                  </h3>
                  <span>
                    На связи в любое время
                  </span>
                </div>
              </div>
              <div
                class="nav__info"
              >
                <div>
                  <p>
                    О компании
                  </p>
                </div>
                <div>
                  <p>
                    Доставка и оплата
                  </p>
                </div>
                <div>
                  <p>
                    Возврат
                  </p>
                </div>
                <div>
                  <p>
                    Контакты
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="header__bot"
        >
          <div
            class="container"
          >
            <div
              class="header__content"
            >
              <span
                class="burger"
              >
                <img
                  alt=""
                  src="/images/burger.svg"
                />
              </span>
              <img
                alt=""
                class="rgb"
                src="/images/RGB.svg"
              />
              <button
                class="header__catalog"
              >
                <img
                  alt=""
                  class="sqg"
                  src="/images/squares_grey.svg"
                />
                Каталог
                <img
                  alt=""
                  class="sq"
                  src="/images/squares.svg"
                />
              </button>
              <div
                class="header__search"
              >
                <input
                  placeholder="Поиск..."
                  type="text"
                />
                <button>
                  <img
                    alt=""
                    class="srg"
                    src="/images/search_grey.svg"
                  />
                  <img
                    alt=""
                    class="sr"
                    src="/images/search.svg"
                  />
                </button>
              </div>
              <div
                class="header__tel"
              >
                <h3>
                  +7 (777) 490-00-91
                </h3>
                <p>
                  время работы: 9:00-20:00
                </p>
                <span>
                  Заказать звонок
                </span>
              </div>
              <img
                alt=""
                class="header__woman"
                src="/images/call_woman.png"
              />
              <button
                class="header__price"
              >
                АДМИН-ПАНЕЛЬ
                <img
                  alt=""
                  src="/images/arrow_down.svg"
                />
              </button>
              <div
                class="header__cart"
              >
                <img
                  alt=""
                  src="/images/cart.svg"
                />
                <span
                  data-testid="количество"
                />
              </div>
              <div
                class="header__cart-check"
              >
                <p>
                  Корзина
                </p>
                <span
                  data-testid="сумма"
                >
                   ₸
                </span>
              </div>
            </div>
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header>
      <div
        class="header__top"
      >
        <div
          class="container"
        >
          <div
            class="nav"
          >
            <div
              class="nav__address"
            >
              <img
                alt=""
                src="/images/marker.svg"
              />
              <div>
                <h3>
                  г. Кокчетав, ул. Ж. Ташенова 129Б
                </h3>
                <span>
                  (Рынок Восточный)
                </span>
              </div>
            </div>
            <div
              class="nav__contact"
            >
              <img
                alt=""
                src="/images/convert.svg"
              />
              <div>
                <h3>
                  opt.sultan@mail.ru
                </h3>
                <span>
                  На связи в любое время
                </span>
              </div>
            </div>
            <div
              class="nav__info"
            >
              <div>
                <p>
                  О компании
                </p>
              </div>
              <div>
                <p>
                  Доставка и оплата
                </p>
              </div>
              <div>
                <p>
                  Возврат
                </p>
              </div>
              <div>
                <p>
                  Контакты
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="header__bot"
      >
        <div
          class="container"
        >
          <div
            class="header__content"
          >
            <span
              class="burger"
            >
              <img
                alt=""
                src="/images/burger.svg"
              />
            </span>
            <img
              alt=""
              class="rgb"
              src="/images/RGB.svg"
            />
            <button
              class="header__catalog"
            >
              <img
                alt=""
                class="sqg"
                src="/images/squares_grey.svg"
              />
              Каталог
              <img
                alt=""
                class="sq"
                src="/images/squares.svg"
              />
            </button>
            <div
              class="header__search"
            >
              <input
                placeholder="Поиск..."
                type="text"
              />
              <button>
                <img
                  alt=""
                  class="srg"
                  src="/images/search_grey.svg"
                />
                <img
                  alt=""
                  class="sr"
                  src="/images/search.svg"
                />
              </button>
            </div>
            <div
              class="header__tel"
            >
              <h3>
                +7 (777) 490-00-91
              </h3>
              <p>
                время работы: 9:00-20:00
              </p>
              <span>
                Заказать звонок
              </span>
            </div>
            <img
              alt=""
              class="header__woman"
              src="/images/call_woman.png"
            />
            <button
              class="header__price"
            >
              АДМИН-ПАНЕЛЬ
              <img
                alt=""
                src="/images/arrow_down.svg"
              />
            </button>
            <div
              class="header__cart"
            >
              <img
                alt=""
                src="/images/cart.svg"
              />
              <span
                data-testid="количество"
              />
            </div>
            <div
              class="header__cart-check"
            >
              <p>
                Корзина
              </p>
              <span
                data-testid="сумма"
              >
                 ₸
              </span>
            </div>
          </div>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
